File: spring-ai-a2a-examples/java/v1/host-agent/src/main/java/io/github/timo/a2a/spring/ai/agent/example/config/RemoteAgentConfig.java
package io.github.timo.a2a.spring.ai.agent.example.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.Map;

@ConfigurationProperties("a2a")
public record RemoteAgentConfig(
    Map<String, RemoteAgentSetting> remoteAgents
) {
}

File: spring-ai-a2a-examples/v1/host-agent/src/main/kotlin/io/github/timo/a2a/spring/ai/agent/example/config/RemoteAgentConfig.kt
package io.github.timo.a2a.spring.ai.agent.example.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("a2a")
data class RemoteAgentConfig(
    val remoteAgents: Map<String, RemoteAgentSetting>
)

data class RemoteAgentSetting(
    val endpoint: String,
    val healthCheckUrl: String? = null,
)

